pkg/prometheus/promcfg_test.go

```diff
@@ -5474,6 +5475,26 @@ scrape_configs: []
 remote_read:
 - url: http://example.com
   remote_timeout: 30s
+  filter_external_labels: false
+`,
+		},
+		{
```

Comment:
we would need more tests:
* with version < 2.34.0 and `filterExternalLabels` defined
* with version >= 2.34.0 and `filterExternalLabels` not defined
* with version >= 2.34.0 and `filterExternalLabels` set to `false`
* with version >= 2.34.0 and `filterExternalLabels` set to `true`

Reply:
Thanks @simonpasquier for the review. I will add the suggested tests ;).